breeze/services/patient_service.py

```diff
@@ -76,11 +89,105 @@ def edit_personal_information(self, user):
         direct_to_dashboard()
     
     def record_mood(self, user):
-        pass
-    
+        clear_screen()
+        print(PATIENT_BANNER_STRING)
+        
+        print("Record Your Mood for the Day")
+        print("Please choose a colour that best describes your mood:")
+        print("[G]reen - Very Happy")
+        print("[L]ight Green - Happy")
+        print("[Y]ellow - Neutral")
+        print("[O]range - Sad")
+        print("[R]ed - Very Sad")
+
+        colour_to_mood = {
+            "g": "Very Happy",
+            "l": "Happy",
+            "y": "Neutral",
+            "o": "Sad",
+            "r": "Very Sad"
+        }
+
+        while True:
+            colour_input = input("Enter the colour code that represents your mood today: ").strip().lower()
+            if colour_input in colour_to_mood:
+                mood_description = colour_to_mood[colour_input]
+                comment = input("Would you like to add any additional comments about your mood today? Type [R] to discard your entry: ").strip()
+                
+                if comment.lower() == 'r':
+                    print_system_message("Mood entry discarded. Returning to dashboard.")
+                    direct_to_dashboard()
+                    return
+
+                date_string = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+                mood_entry = {
+                    "mood": mood_description,
+                    "comment": comment,
+                    "datetime": date_string
+                }
+
+                if hasattr(user, "add_mood_entry"):
+                    user.add_mood_entry(mood_description, comment, date_string)
+
+                self.auth_service.save_data_to_file()
+
+                print_system_message("Mood recorded successfully!")
+                direct_to_dashboard()  
+                return
+            else:
+                print_system_message("Invalid colour entered. Please choose from Green, Light Green, Yellow, Orange, or Red.")
+
     def enter_journaling(self, user):
-        pass
-    
+        clear_screen()
+        print(PATIENT_BANNER_STRING)
+        print(f'Hi {user.get_username()} !')
+        print_system_message('Write your journal entry below, or enter [R] at any time to return to the previous page without saving')
+        journal_title = input('What is the title of your entry? \n').strip()
+        if return_to_previous(journal_title, 'r'):
+            return
+        if journal_title.strip() == '':
+            print_system_message('Your title is empty! Returning to dashboard...')
+            time.sleep(2)
+            return
+        journal_body = input('Write your journal entry here: \n').strip()
+        if self.return_to_previous(journal_body, 'r'):
```

Comment:
An error popped up saying it couldn’t find `self.return_to_previous()`. Since `return_to_previous()` is no longer a method of this instance, you can use `return_to_previous` directly instead.

Reply:
Fixed, I had some problems with the git rebase command with having to address a lot of merge conflicts and I think this slipped through the cracks. Thanks though, this should be good now.